src/overlays/actors/ovl_Obj_Boat/z_obj_boat.c

```diff
@@ -28,31 +32,164 @@ const ActorInit Obj_Boat_InitVars = {
     (ActorFunc)ObjBoat_Draw,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80B9B680[] = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneScale, 1000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 1000, ICHAIN_STOP),
 };
 
-#endif
-
-extern InitChainEntry D_80B9B680[];
-
-extern UNK_TYPE D_06007630;
-extern UNK_TYPE D_06009A88;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Boat/func_80B9AF50.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Boat/ObjBoat_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Boat/ObjBoat_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Boat/func_80B9B124.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Boat/ObjBoat_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Boat/func_80B9B428.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Boat/ObjBoat_Draw.s")
+s16 func_80B9AF50(ObjBoat* this, Vec3f* arg0) {
+    s16 yaw;
+    Vec3s* temp = &this->unk164[(s32)this->unk15C];
+
+    Math_Vec3s_ToVec3f(arg0, &temp[this->unk15D]);
+    yaw = Math_Vec3f_Yaw(&this->dyna.actor.world.pos, arg0);
+
+    return ((this->unk15D > 0) ? yaw : (yaw + 0x8000));
+}
+
+void ObjBoat_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad[2];
+    Path* path;
+    ObjBoat* this = THIS;
+    Vec3f sp24;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    DynaPolyActor_Init(&this->dyna, 3);
+    DynaPolyActor_LoadMesh(globalCtx, &this->dyna, &object_kaizoku_obj_Colheader_009A88);
+    if (this->dyna.actor.params < 0) {
+        this->dyna.actor.update = (void (*)(Actor*, GlobalContext*))func_80B9B428;
```

Comment:
Is this cast here necessary?

Reply:
this is what happens if i take it out, i wonder if there's some actionFunc that i need to add to the header, but the only padding variable there is 1 byte...

warning: assignment to ‘ActorFunc’ {aka ‘void (*)(struct Actor *, struct GlobalContext *)’} from incompatible pointer type ‘void (*)(ObjBoat *, GlobalContext *)’ {aka ‘void (*)(struct ObjBoat *, struct GlobalContext *)’} [-Wincompatible-pointer-types]

Reply:
`func_80B9B428` should be changed to take the base `Actor* thisx` instead of `ObjBoat* this`, than the cast would no longer be needed.

Reply:
This still needs doing, looks like `func_80B9B428` is the update used by the boat in cutscenes.